src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -833,31 +816,21 @@ internal void TakeSnapshot(bool outputDelta)
                 {
                     if (outputDelta)
                     {
-                        // TODO:
```

Comment:
Finally, we get to remove some of the oldest TODOs in the critical path 👍

Reply:
@cijothomas This hasn't really changed the way it's reading the double value in that it still uses the same trick of using CompareExchange (it just has updated the parameters, instead of comparing with `double.NegativeInfinity` it's comparing with `double.Nan`)

Reply:
The TODO wasn't about changing anything. It was just to confirm that this is correct with someone more knowledgeable about this area!

Reply:
I did make two tweaks by the way.

```diff
-Interlocked.CompareExchange(ref this.runningValue.AsDouble, 0.0, double.NegativeInfinity)
+Interlocked.CompareExchange(ref location, double.NaN, double.NaN);
```

Before we looked for `double.NegativeInfinity` and would switch to `0.0` in that case. Since I was making a general helper method, that seemed a bit risky to me. I know it isn't likely but who knows how this helper will be used in the future and if `double.NegativeInfinity` might be meaningful to someone. I decided to use `double.NaN` because that seemed even less likely and then also we put it back now if we happen to see it so no harm no foul.